### PR TITLE
Cherrypicking based on branch labels added to the parent PR

### PR DESCRIPTION
**Pull request by jyejare**
Since we are moving to a more complex auto cherrypicking which is upto the depth of `n-3`. As a resultmanually adding the `CherryPick` label onto the Cherrypicked PRs to get them cherrypicked to zStream branches would be a pathetic nagger for authors.

So we are here with the solution:
- Add all the `zStream Version` labels to your master PR that you want to see the master changes cherrypicked to.
- Autocherrypick GHA will create the cherrypicked PRs for all the labeled versions in master PR
- The individual label based failure notifications would be sent to GChat
- The individual label based issue would be created in Github for failed cherrypicks

To-DO (But does not blocks this PR from getting merged):

- [ ] Add support in GChat Notification GHA for showing the branch label

### CODE REVIEW

File: .github/workflows/auto_cherry_pick.yml [hunk 1]
```diff
@@ -6,20 +6,15 @@ on:
...
-    outputs:
-      previous_branch: ${{ steps.set-branch.outputs.previous_branch }}
-    steps:
-      - uses: actions/checkout@v3
-        with:
-          fetch-depth: 0
-      - id: set-branch
-        run: echo "previous_branch=$(if [ $GITHUB_BASE_REF  == 'master' ]; then echo $(git branch -rl 'origin/6.*.z' --format='%(refname:lstrip=-1)' | sort --version-sort | tail -n1); else echo '6.'$(($(echo $GITHUB_BASE_REF | cut -d. -f2) - 1))'.z'; fi)" >> $GITHUB_OUTPUT
 
+  # Pre-Requisites for Auto Cherrypicking
```
Comment (omkarkhatavkar) [inline .github/workflows/auto_cherry_pick.yml:17]: I don't think so it is prerequisite for auto cherrypicking, we could say may be auto merging auto-cherrypick PR's

File: .github/workflows/auto_cherry_pick.yml [hunk 2]
```diff
@@ -6,20 +6,15 @@ on:
     types:
       - closed
 
+# Github & Parent PR Env vars
+env:
+  assignee: ${{ github.event.pull_request.assignee.login }}
+  title: ${{ github.event.pull_request.title }}
+  number: ${{ github.event.number }}
+
```
Comment (omkarkhatavkar) [inline .github/workflows/auto_cherry_pick.yml:14]: nice!

**Review by omkarkhatavkar — APPROVED** [review]
ACK

File: .github/workflows/auto_cherry_pick.yml [hunk 3]
```diff
@@ -66,6 +67,19 @@ jobs:
           message: ${{ needs.find-the-parent-prt-comment.outputs.prt_comment }}
           repo-token: ${{ secrets.CHERRYPICK_PAT }}
 
+      - name: is autoMerging enabled for Auto-Cherry-Pick PR's
```
Comment (omkarkhatavkar) [inline .github/workflows/auto_cherry_pick.yml:70]: @jyejare added this for label `AutoMergeCherryPick`

Comment (jyejare) [reply .github/workflows/auto_cherry_pick.yml:70]: @omkarkhatavkar Added a label `AutoMerge_Cherry_Picked` different what you suggested above to keep all cherrypick labels consistent.

Comment (omkarkhatavkar) [reply .github/workflows/auto_cherry_pick.yml:70]: Done !

File: .github/workflows/auto_cherry_pick.yml (hunk 1 above)
Comment (jyejare) [reply .github/workflows/auto_cherry_pick.yml:17]: Here I mean all the pre-requisites needed for the entire auto-cherry-pick job.

File: .github/workflows/auto_cherry_pick.yml [hunk 4]
```diff
@@ -66,6 +67,19 @@ jobs:
...
+        if: contains(github.event.pull_request.labels.*.name, 'AutoMerge_Cherry_Picked')
```
Comment (jyejare) [inline .github/workflows/auto_cherry_pick.yml:71]: You should also check if the result of the step is successful.

File: .github/workflows/auto_cherry_pick.yml (hunk 3 above)
Comment (jyejare) [inline .github/workflows/auto_cherry_pick.yml:70]: `name: is autoMerging enabled for Auto CherryPicked PRs ?`

**Review by jyejare — COMMENTED** [review]
@omkarkhatavkar Some comments to address.

**Comment by jyejare** [comment]
@omkarkhatavkar Looks like the checks are failing, please handle them as well.

**Comment by omkarkhatavkar** [comment]
@jyejare We might need to remove the cherry-pick code from the 6.11.z as well.

**Comment by jyejare** [comment]
@omkarkhatavkar Not required we can keep it for future needs if any! There is no harm in keeping it there. But until its not required/suggested lets not cherrypick things to 6.10.z but we have to cherrypick to 6.11.z

**Pull request merged**